File: animind/src/screens/MemoryScreen.js
import React, { useState, useCallback } from 'react';
import {
  View,
  Text,
  FlatList,
  TouchableOpacity,
  StyleSheet,
  Alert,
} from 'react-native';
import { useFocusEffect } from '@react-navigation/native';
import { getAllMemories, deleteMemory, clearAllMemories } from '../db/database';

export default function MemoryScreen() {
  const [memories, setMemories] = useState([]);

  const loadMemories = useCallback(async () => {
    const mems = await getAllMemories();
    setMemories(mems);
  }, []);

  useFocusEffect(
    useCallback(() => {
      loadMemories();
    }, [loadMemories])
  );

  const handleDelete = (id) => {
    Alert.alert('Delete Memory', 'Remove this memory?', [
      { text: 'Cancel', style: 'cancel' },
      {
        text: 'Delete',
        style: 'destructive',
        onPress: async () => {
          await deleteMemory(id);
          loadMemories();
        },
      },
    ]);
  };

  const handleClearAll = () => {
    Alert.alert(
      'Clear All Memories',
      'This will erase everything Pixel knows about you. Are you sure?',
      [
        { text: 'Cancel', style: 'cancel' },
        {
          text: 'Clear All',
          style: 'destructive',
          onPress: async () => {
            await clearAllMemories();
            loadMemories();
          },
        },
      ]
    );
  };

  const formatDate = (timestamp) => {
    const d = new Date(timestamp);
    return d.toLocaleDateString() + ' ' + d.toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' });
  };

  const importanceBar = (score) => {
    const filled = Math.round(score * 5);
    return '●'.repeat(filled) + '○'.repeat(5 - filled);
  };

  const renderMemory = ({ item }) => (
    <TouchableOpacity
      style={styles.memoryCard}
      onLongPress={() => handleDelete(item.id)}
      activeOpacity={0.7}
    >
      <Text style={styles.memoryContent}>{item.content}</Text>
      <View style={styles.memoryMeta}>
        <Text style={styles.memoryDate}>{formatDate(item.timestamp)}</Text>
        <Text style={styles.importanceScore}>
          {importanceBar(item.importance_score)}
        </Text>
      </View>
    </TouchableOpacity>
  );

  return (
    <View style={styles.container}>
      {/* Header */}
      <View style={styles.header}>
        <View>
          <Text style={styles.title}>Pixel's Memories</Text>
          <Text style={styles.subtitle}>
            {memories.length} memor{memories.length === 1 ? 'y' : 'ies'} stored
          </Text>
        </View>
        {memories.length > 0 && (
          <TouchableOpacity style={styles.clearButton} onPress={handleClearAll}>
            <Text style={styles.clearButtonText}>Clear All</Text>
          </TouchableOpacity>
        )}
      </View>

      {/* Memory list */}
      <FlatList
        data={memories}
        renderItem={renderMemory}
        keyExtractor={(item) => item.id.toString()}
        contentContainerStyle={styles.list}
        ListEmptyComponent={
          <View style={styles.emptyState}>
            <Text style={styles.emptyEmoji}>🧠</Text>
            <Text style={styles.emptyTitle}>No memories yet</Text>
            <Text style={styles.emptyText}>
              Chat with Pixel to start building memories.{'\n'}
              Key facts will be extracted and stored here.
            </Text>
          </View>
        }
      />

      <Text style={styles.hint}>Long-press a memory to delete it</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FAF3E0',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 16,
    paddingTop: 8,
    borderBottomWidth: 1,
    borderBottomColor: '#E8DCC8',
    backgroundColor: '#FFF8E7',
  },
  title: {
    fontSize: 20,
    fontWeight: '700',
    color: '#5D4037',
  },
  subtitle: {
    fontSize: 13,
    color: '#A0896C',
    marginTop: 2,
  },
  clearButton: {
    backgroundColor: '#FF5252',
    paddingHorizontal: 14,
    paddingVertical: 8,
    borderRadius: 8,
  },
  clearButtonText: {
    color: '#FFF',
    fontSize: 13,
    fontWeight: '700',
  },
  list: {
    padding: 16,
    paddingBottom: 40,
  },
  memoryCard: {
    backgroundColor: '#FFF',
    borderRadius: 12,
    padding: 14,
    marginBottom: 10,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 1 },
    shadowOpacity: 0.06,
    shadowRadius: 4,
    elevation: 2,
  },
  memoryContent: {
    fontSize: 15,
    color: '#333',
    lineHeight: 21,
  },
  memoryMeta: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginTop: 8,
  },
  memoryDate: {
    fontSize: 11,
    color: '#A0896C',
  },
  importanceScore: {
    fontSize: 10,
    color: '#7C4DFF',
    letterSpacing: 2,
  },
  emptyState: {
    alignItems: 'center',
    paddingTop: 60,
  },
  emptyEmoji: {
    fontSize: 48,
    marginBottom: 16,
  },
  emptyTitle: {
    fontSize: 18,
    fontWeight: '700',
    color: '#5D4037',
    marginBottom: 8,
  },
  emptyText: {
    fontSize: 14,
    color: '#A0896C',
    textAlign: 'center',
    lineHeight: 20,
  },
  hint: {
    textAlign: 'center',
    fontSize: 11,
    color: '#C0A888',
    paddingBottom: 16,
  },
});

File: animind_flutter/lib/screens/memory_screen.dart
import 'package:flutter/material.dart';
import 'package:intl/intl.dart';
import '../models/memory.dart';
import '../services/database_service.dart';

class MemoryScreen extends StatefulWidget {
  const MemoryScreen({super.key});

  @override
  State<MemoryScreen> createState() => _MemoryScreenState();
}

class _MemoryScreenState extends State<MemoryScreen> with AutomaticKeepAliveClientMixin {
  List<Memory> _memories = [];

  @override
  bool get wantKeepAlive => true;

  @override
  void initState() {
    super.initState();
    _loadMemories();
  }

  @override
  void didChangeDependencies() {
    super.didChangeDependencies();
    _loadMemories();
  }

  Future<void> _loadMemories() async {
    final mems = await DatabaseService.getAllMemories();
    if (mounted) setState(() => _memories = mems);
  }

  Future<void> _deleteMemory(int id) async {
    final confirmed = await showDialog<bool>(
      context: context,
      builder: (ctx) => AlertDialog(
        title: const Text('Delete Memory'),
        content: const Text('Remove this memory?'),
        actions: [
          TextButton(onPressed: () => Navigator.pop(ctx, false), child: const Text('Cancel')),
          TextButton(
            onPressed: () => Navigator.pop(ctx, true),
            child: const Text('Delete', style: TextStyle(color: Colors.red)),
          ),
        ],
      ),
    );
    if (confirmed == true) {
      await DatabaseService.deleteMemory(id);
      _loadMemories();
    }
  }

  Future<void> _clearAll() async {
    final confirmed = await showDialog<bool>(
      context: context,
      builder: (ctx) => AlertDialog(
        title: const Text('Clear All Memories'),
        content: const Text('This will erase everything Pixel knows about you. Are you sure?'),
        actions: [
          TextButton(onPressed: () => Navigator.pop(ctx, false), child: const Text('Cancel')),
          TextButton(
            onPressed: () => Navigator.pop(ctx, true),
            child: const Text('Clear All', style: TextStyle(color: Colors.red)),
          ),
        ],
      ),
    );
    if (confirmed == true) {
      await DatabaseService.clearAllMemories();
      _loadMemories();
    }
  }

  String _formatDate(String timestamp) {
    final dt = DateTime.parse(timestamp);
    return DateFormat('dd/MM/yyyy HH:mm').format(dt);
  }

  String _importanceBar(double score) {
    final filled = (score * 5).round();
    return '●' * filled + '○' * (5 - filled);
  }

  @override
  Widget build(BuildContext context) {
    super.build(context);

    return Scaffold(
      appBar: AppBar(title: const Text('🧠 Memories')),
      body: Column(
        children: [
          // Header
          Container(
            padding: const EdgeInsets.all(16),
            decoration: const BoxDecoration(
              color: Color(0xFFFFF8E7),
              border: Border(bottom: BorderSide(color: Color(0xFFE8DCC8))),
            ),
            child: Row(
              children: [
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      const Text(
                        "Pixel's Memories",
                        style: TextStyle(
                          fontSize: 20,
                          fontWeight: FontWeight.w700,
                          color: Color(0xFF5D4037),
                        ),
                      ),
                      const SizedBox(height: 2),
                      Text(
                        '${_memories.length} memor${_memories.length == 1 ? 'y' : 'ies'} stored',
                        style: const TextStyle(
                          fontSize: 13,
                          color: Color(0xFFA0896C),
                        ),
                      ),
                    ],
                  ),
                ),
                if (_memories.isNotEmpty)
                  FilledButton.tonal(
                    onPressed: _clearAll,
                    style: FilledButton.styleFrom(
                      backgroundColor: const Color(0xFFFF5252),
                      foregroundColor: Colors.white,
                    ),
                    child: const Text('Clear All'),
                  ),
              ],
            ),
          ),

          // List
          Expanded(
            child: _memories.isEmpty
                ? _buildEmptyState()
                : RefreshIndicator(
                    onRefresh: _loadMemories,
                    child: ListView.builder(
                      padding: const EdgeInsets.all(16),
                      itemCount: _memories.length,
                      itemBuilder: (context, index) =>
                          _buildMemoryCard(_memories[index]),
                    ),
                  ),
          ),

          if (_memories.isNotEmpty)
            const Padding(
              padding: EdgeInsets.only(bottom: 12),
              child: Text(
                'Long-press a memory to delete it',
                style: TextStyle(fontSize: 11, color: Color(0xFFC0A888)),
              ),
            ),
        ],
      ),
    );
  }

  Widget _buildEmptyState() {
    return Center(
      child: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
          const Text('🧠', style: TextStyle(fontSize: 48)),
          const SizedBox(height: 16),
          const Text(
            'No memories yet',
            style: TextStyle(
              fontSize: 18,
              fontWeight: FontWeight.w700,
              color: Color(0xFF5D4037),
            ),
          ),
          const SizedBox(height: 8),
          Text(
            'Chat with Pixel to start building memories.\nKey facts will be extracted and stored here.',
            textAlign: TextAlign.center,
            style: TextStyle(
              fontSize: 14,
              color: Colors.brown.shade300,
              height: 1.5,
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildMemoryCard(Memory memory) {
    return GestureDetector(
      onLongPress: () => _deleteMemory(memory.id!),
      child: Card(
        margin: const EdgeInsets.only(bottom: 10),
        elevation: 1,
        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
        color: Colors.white,
        child: Padding(
          padding: const EdgeInsets.all(14),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Text(
                memory.content,
                style: const TextStyle(
                  fontSize: 15,
                  color: Color(0xFF333333),
                  height: 1.4,
                ),
              ),
              const SizedBox(height: 8),
              Row(
                mainAxisAlignment: MainAxisAlignment.spaceBetween,
                children: [
                  Text(
                    _formatDate(memory.timestamp),
                    style: const TextStyle(
                      fontSize: 11,
                      color: Color(0xFFA0896C),
                    ),
                  ),
                  Text(
                    _importanceBar(memory.importanceScore),
                    style: const TextStyle(
                      fontSize: 10,
                      color: Color(0xFF7C4DFF),
                      letterSpacing: 2,
                    ),
                  ),
                ],
              ),
            ],
          ),
        ),
      ),
    );
  }
}
